File: js_console/GuessingGame.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function random(min, max) {
    return Math.trunc(Math.random() * (max - min) + min);
}

rangeStart = 1;
rangeEnd = 10;
number = random(rangeStart, rangeEnd);

console.log(`Pick a number between ${rangeStart} and ${rangeEnd}`);

notGuessed = true;
guessCount = 0;
numberGuessed = 0;
rl.on('line', function (userEntered) {
    try {
        numberGuessed = parseInt(userEntered)
        guessCount += 1;
    } catch (err) {
        console.log("Invalid number. Try again.");
        return;
    }
    if (numberGuessed == number) {
        console.log(numberGuessed + ` was the number. It took you ${guessCount} guesses`);
        process.exit(0);
    }
    else {
        if (numberGuessed < rangeStart || numberGuessed > rangeEnd) {
            console.log(`That number is not between ${rangeStart} and ${rangeEnd}`);
            return;
        }
        if (numberGuessed < number) {
            console.log("That number is too low");
            return;
        }
        if (numberGuessed > number) {
            console.log("That number is too high");
            return;
        }
    }
});

File: py_console/GuessingGame.py
import random

random.seed()

rangeStart = 1
rangeEnd = 20
number = random.randint(rangeStart, rangeEnd)

print("Pick a number between {0} and {1}".format(rangeStart, rangeEnd))

notGuessed = True
guessCount = 0
while notGuessed:
    userEntered = input()
    try:
        numberGuessed = int(userEntered)
        guessCount += 1
    except ValueError:
        print("Invalid number. Try again.")
        continue
    if numberGuessed == number:
        print("Correct! {0} was the number. It took you {1} guesses".format(numberGuessed, guessCount))
        notGuessed = False
    else:
        if numberGuessed < rangeStart or numberGuessed > rangeEnd:
            print("That number is not between {0} and {1}".format(rangeStart, rangeEnd))
            continue
        if numberGuessed < number:
            print("That number is too low. Try again.")
            continue
        if numberGuessed > number:
            print("That number is too high. Try again.")
            continue